### PR TITLE
Sw primary button

### PR DESCRIPTION
**Pull request by misspia**
Based on https://github.com/SnapWars/SnapWar/pull/12

- No UI changes
- Create a base button and primary button class (will add more to them as we go and figure out what we need)
- Replace the auth view button with a `SWPrimaryButton`

### CODE REVIEW

File: SnapWar/ViewControllers/SWAuthViewController.swift [hunk 1]
```diff
@@ -39,15 +39,13 @@ class SWAuthViewController: UIViewController, UITextFieldDelegate {
         passwordField.tag = 1
         formContainer.addArrangedSubview(passwordField)
         
-        loginButton.setTitle("Login", for: .normal)
-        loginButton.backgroundColor = SWColorType.primary
         loginButton.addTarget(self, action: #selector(handleSubmit), for: .touchUpInside)
-        setLoginButtonState()
+        loginButton.isEnabled = isFormValid()
```
Comment (davidozhang) [inline SnapWar/ViewControllers/SWAuthViewController.swift]: hmmm i feel like you should wrap this line inside a method

**Review by davidozhang — APPROVED** [review]
code looks good, what's the diff between primary and base button?

File: SnapWar/ViewControllers/SWAuthViewController.swift (hunk 1 above)
Comment (misspia) [reply SnapWar/ViewControllers/SWAuthViewController.swift]: oki

**Comment by misspia** [comment]
Primary button will use be an extension of the base button with its own styles :)

**Pull request merged**